helix-core/src/test/java/org/apache/helix/TestHelper.java

```diff
@@ -803,19 +803,15 @@ public interface Verifier {
   }
 
   public static boolean verify(Verifier verifier, long timeout) throws Exception {
-    long start = System.currentTimeMillis();
-    do {
-      boolean result = verifier.verify();
-      boolean isTimedout = (System.currentTimeMillis() - start) > timeout;
-      if (result || isTimedout) {
-        if (isTimedout && !result) {
-          LOG.error("verifier time out, consider try longer timeout, stack trace{}",
-              Arrays.asList(Thread.currentThread().getStackTrace()));
-        }
-        return result;
-      }
-      Thread.sleep(50);
-    } while (true);
+    try {
+      with().pollInterval(50, TimeUnit.MILLISECONDS).atMost(timeout, TimeUnit.MILLISECONDS).await()
+          .until(verifier::verify);
+      return verifier.verify();
+    } catch (Exception ex) {
+      LOG.error("verifier time out, consider try longer timeout, stack trace{}",
```

Comment:
I think when verify times out, it will throw exception and log stack trace. Catch - log and re throw is not needed.

Reply:
In the previous implementation it looks like upon timeout, a false is returned upon the expiration of the timeout (preceded by a log err indicating that the verification has timed out). In this iteration, we are throwing the exception back but the catch-rethrow is just to render the log line.

Reply:
Thanks for explain. I feel like there is no need to log here. Exception will print stack trace.

Reply:
Great. Removing the catch-rethrow